src/util/amount.rs

```diff
@@ -43,6 +47,8 @@ impl Denomination {
             Denomination::Bitcoin => -8,
             Denomination::MilliBitcoin => -5,
             Denomination::MicroBitcoin => -2,
+            Denomination::NanoBitcoin => 1,
+            Denomination::PicoBitcoin => 4,
```

Comment:
I really sorry, but I do not get it. What is "picoBitcoin"? It should be 10^-12 and 3 orders of magnitude less than nanoBitcoin - i.e. less than satoshi for two orders of magnitude.

```suggestion
            Denomination::PicoBitcoin => -1,
```

Reply:
The PR is correct I believe, lets see if I can clarify. These numbers go the opposite way to intuition, see the comment
```
/// The number of decimal places more than a satoshi.
```
So BTC is 8 decimal places _less_ than a satoshi (hence -8), PicoBitcoin is 4 decimal places _more_ than a satoshi.

Does that help?

Reply:
Yeah I was confused too when I reviewed it. My thinking was assuming `MilliBitcoin` and `MicroBitcoin` were correct, logically the number should be increased by three for each subsequent denomination: -2 + 3 == 1, 1 + 3 == 4.

Reply:
right since the current precision is using `satoshi` as a base unit to do comparison but it's not the canonical smallest unit so we have both negative and positive number(that's the probably the part confuse people?)

I also found the function name `precision` not too clear to me. (In decimal, precision means total number of digit).

If we can treat `Bitcoin as canonical largest unit`(I doubt we need other unit larger than Bitcoin while total cap <21million BTC ?), maybe we can change this function to `power` (power of 10 relative to Bitcoin unit).  and then it will be a decreasing negative number like(Here I think it's more natural to order by power)


```
            Denomination::Bitcoin => 0,
            Denomination::MilliBitcoin => -3,
            Denomination::MicroBitcoin => -6,
            Denomination::Bit => -6,
            Denomination::Satoshi => -8,
            Denomination::NanoBitcoin => -9,
            Denomination::MilliSatoshi => -11,
            Denomination::PicoBitcoin => -12,
   ```

Reply:
Probably we may do this as a follow-up PR. I will merge the current PR since it already have reviews and solves at least one problem.